### PR TITLE
Adds Config API

### PR DESCRIPTION
**Pull request by danehans**
Adds Config API and tooling in support of #95.

xref: https://github.com/envoyproxy/gateway/issues/29

Signed-off-by: danehans <daneyonhansen@gmail.com>

### CODE REVIEW

File: Makefile.targets.mk [hunk 1]
```diff
@@ -11,6 +11,42 @@ IMAGE ?= ${REGISTRY}/gateway-dev
 REV=$(shell git rev-parse --short HEAD)
 # Tag is the tag to use for build and push image targets.
 TAG ?= $(REV)
+# ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
```
[the remark this answers is not in the document]
Comment (danehans) [reply Makefile.targets.mk:14]: Based on [this vote](https://github.com/envoyproxy/gateway/issues/90#issuecomment-1169338289), the community has decided to focus on using the "local dev" approach and no longer support the `MAKE_IN_DOCKER` approach. @LukeShu is assigned https://github.com/envoyproxy/gateway/issues/124 and will have a PR submitted to transition the current Docker-based linters. With that said, IMHO the targets added by this PR should be acceptable.

**Review by youngnick — COMMENTED** [review]
I think the structs that are the core here look great to get us started.

Perhaps in a followup, we could consider putting a .gitattributes into the root, so that the generated files are shown as collapsed automatically in the Github UI?

Gateway API does this in https://github.com/kubernetes-sigs/gateway-api/blob/main/.gitattributes .

**Review by youngnick — DISMISSED** [review]
Oops, meant to approve. The other comment is non-blocking.

**Review by alexgervais — APPROVED** [review]
🙌

**Review by alexgervais — APPROVED** [review]

**Pull request merged**